java/src/jmri/InstanceManager.java

```diff
@@ -709,7 +714,74 @@ static public void setSensorManager(SensorManager p) {
      * Default constructor for the InstanceManager.
      */
     public InstanceManager() {
-        // do nothing
+        ServiceLoader.load(InstanceInitializer.class).forEach((provider) -> {
+            provider.getInitalizes().forEach((cls) -> {
+                this.initializers.put(cls, provider);
```

Comment:
Just a comment about using clearAll so its in the record:

I'm not recreating the InstanceManager because I want ```dispose()``` to be called against ```Disposable``` instances between tests (in case they are doing something that could cause problems if a single managed object were initialized twice without first being disposed).